### PR TITLE
[gsoc2020/chores] Merge django-netjsongraph & code cleanup

### LINKED ISSUE #58: [gsoc2020]  Merge django-netjsongraph into openwisp-network-topology
- [x] Merge django-netjsongraph into openwisp-network-topology
- [x] Migrate the django settings of each module so that the naming is consistent. (eg: `NETJSONCONFIG_[SETTING_NAME]` -> `OPENWISP_CONTROLLER_[SETTING_NAME]`)
- [x] Keep backward compatibility for setting names but log a warning to encourage the users to upgrade the name of the setting to the new name.
- [x] Implement as easy as possible extensibility of Django modules.
- [x] Add sample_app for testing extensibility with some changes.
- [x] Update documentation

### LINKED ISSUE #59: [api] Add v1 prefix to /api/ > /api/v1/
Similar to https://github.com/openwisp/openwisp-controller/issues/199.

@atb00ker can you do this as part of the clean up work in this module pelase?
If yes, please assign it and put it into the board.

### LINKED ISSUE #61: [api] NetworkGraphSerializer.Meta.model
@atb00ker, can you please take care of this when introducing swappable and merging? https://github.com/openwisp/openwisp-network-topology/commit/a0c5d59e009b32fcf7355dc02aed00473dd3f299

### PR DESCRIPTION
**Pull request by atb00ker**
Closes #58
Closes #59
Closes #61

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -14,21 +15,49 @@ openwisp-network-topology
...
+* **receive topology** from multiple nodes
+* **topology history**: allows saving daily snapshots of each topology that can be viewed in the frontend
+
+Project goals
+-------------
+
+* make it easy to visualize network topology data for the formats supported by `netdiff <https://github.com/openwisp/netdiff>`_
+* expose topology data via RESTful resources in *NetJSON NetworkGraph* format
+* make it easy to integrate in larger django projects to improve reusability
+* make it easy to extend its models by providing abstract models (**needs improvement in this point**)
```
Comment (nemesisdesign) [inline README.rst:47]: I think we can remove "needs improvement in this point"

File: README.rst [hunk 2]
```diff
@@ -219,6 +180,583 @@ Run qa tests:
...
+If you extend these views, remember to use these views in the
+``api/urls.py``.
+
+2. Extending the Visualizer views
+#################################
+
+Similar to API views, visualizer views are only required to be extended
+when you want to make changes in the Visualizer.
+Please refer to `sample_network_topology/visualizer/views.py
+<tests/openwisp2/sample_network_topology/visualizer/views.py>`_
```
Comment (nemesisdesign) [inline README.rst:755]: use absolute paths please

File: README.rst (hunk 1 above)
Comment (atb00ker) [reply README.rst:47]: Lol, yes, I'll remove that! :smile:

File: openwisp_network_topology/urls.py [hunk 3]
```diff
@@ -5,6 +5,6 @@
 
 urlpatterns = [
     url(r'^accounts/', include('openwisp_users.accounts.urls')),
-    url(r'^api/', include(api)),
+    url(r'^api/v1/', include(api)),
```
Comment (nemesisdesign) [inline openwisp_network_topology/urls.py:8]: this is a backward incompatible change, can you add a note in CHANGES.rst please?

File: README.rst [hunk 4]
```diff
@@ -1,34 +1,80 @@
...
-.. figure:: docs/images/visualizer.png
-   :align: center
-   :alt:
+.. image:: https://img.shields.io/badge/code%20style-black-000000.svg
+   :target: https://pypi.org/project/black/
+   :alt: code style: black
 
-------------
+.. image:: https://github.com/openwisp/openwisp-network-topology/raw/master/docs/demo_network_topology.gif
+   :alt: Features Highlights
```
Comment (nemesisdesign) [inline README.rst:36]: Can you add the openwisp logo as it's done here please?
https://github.com/openwisp/openwisp-utils/blob/master/README.rst

Comment (atb00ker) [reply README.rst:39]: Done it but the logo and the git together are not put in other repositories, like django-x509, openwisp-ipam should the logo be added in those as well?

Comment (nemesisdesign) [reply README.rst:39]: Yes, I think a structure that is visually appealing is:

1. GIF
2.short description (you can copy and paste what's in the repo description above or write a better one if you have a better idea).
3 openwisp logo

I think it communicates that the project is being taken care of.
The only missing step in some modules is to release and release more often.
Once we get to release point version (1.0.0, 2.0.0) the process will be complete.

All the steps we are doing are aimed at making each module look something other developers and users can rely on.

Comment (nemesisdesign) [reply README.rst:35]: eg: https://github.com/openwisp/openwisp-network-topology/pull/63/commits/09c52c863ee1eabbd3e12f918f0e61748937274d

**Comment by atb00ker** [comment]
Rebased again! :smile:

**Comment by atb00ker** [comment]
Travis job (Badge not appearing): https://travis-ci.org/github/openwisp/openwisp-network-topology/builds/695188585

**Pull request merged**